### PR TITLE
Place Istio validation logging under 'validation' scope

### PR DESCRIPTION
**Pull request by esnible**
Resolve https://github.com/istio/istio/issues/17842 by placing Istio CRD validation logging under the 'validation' log scope and making that scope default to ERROR only.

Capitalize Envoy in log messages and Error returns to comply with https://istio.io/about/contribute/style-guide/ for any string that might surface to a user.

### CODE REVIEW

File: pkg/config/xds/xds.go [hunk 1]
```diff
@@ -56,11 +56,11 @@ func BuildXDSObjectFromStruct(applyTo networking.EnvoyFilter_ApplyTo, value *typ
 	case networking.EnvoyFilter_HTTP_ROUTE:
 		obj = &route.Route{}
 	default:
-		return nil, fmt.Errorf("envoy filter: unknown object type for applyTo %s", applyTo.String())
+		return nil, fmt.Errorf("Envoy filter: unknown object type for applyTo %s", applyTo.String()) // nolint: golint,stylecheck
```
Comment (howardjohn) [inline pkg/config/xds/xds.go]: errors are not supposed to start with capital letters, even if its a proper noun I think. Hence the lint errors

Comment (esnible) [reply pkg/config/xds/xds.go]: Proper nouns should be capitalized, according to Go standards:
https://github.com/golang/go/wiki/CodeReviewComments#error-strings

This isn't just an Istio style issue.   IBM (and most companies) have a policies about trademarks.  Every year I must personally affirm that I will "Use IBM’s and other companies’ trademarks properly and acknowledge the trademarks of others, when appropriate."  If I see trademarks being misused when creating a PR I try to fix them.

Comment (geeknoid) [reply pkg/config/xds/xds.go]: Can you just change the phrasing to avoid the problem?

"unknown object type for applyTo field in Envoy filter:"

Comment (esnible) [reply pkg/config/xds/xds.go]: @geeknoid `lint_istio` is passing, it isn't a problem.  The `// nolint` declarations are being used as intended to signal that we really did intend to start the Error with a capital letter.

**Pull request merged**